src/ast/codegen_llvm.cpp

```diff
@@ -1611,7 +1611,16 @@ void CodegenLLVM::visit(FieldAccess &acc)
         raw = b_.CreateLoad(dst);
         b_.CreateLifetimeEnd(dst);
       }
-      Value *shifted = b_.CreateLShr(raw, field.bitfield.access_rshift);
+      size_t rshiftbits;
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+      rshiftbits = field.bitfield.access_rshift;
+#else
+      // Number of bytes to shift
```

Comment:
```suggestion
      // Number of bits to shift
```

Reply:
but i'd leave the comment out, or explain the logic behind how this shifting works

Reply:
Ok, Sure. I will explain with an example in the commit message then. Thanks.

Reply:
Added in the commit message. Thanks